src/fpm/cmd/new.f90

```diff
@@ -98,9 +98,6 @@ subroutine cmd_new(settings)
     ! like realpath() or getcwd().
     bname=basename(settings%name)
 
-    ! create NAME/.gitignore file
-    call warnwrite(join_path(settings%name, '.gitignore'), ['build/*'])
```

Comment:
I'm currently using the following `.gitignore` in my projects (GitHub's default for Fortran projects), I still think this would be a useful file to always have around.

```gitignore
# Prerequisites
*.d

# Compiled Object files
*.slo
*.lo
*.o
*.obj

# Precompiled Headers
*.gch
*.pch

# Compiled Dynamic libraries
*.so
*.dylib
*.dll

# Fortran module files
*.mod
*.smod

# Compiled Static libraries
*.lai
*.la
*.a
*.lib

# Executables
*.exe
*.out
*.app
```

Ignoring build directories can be tricky, the original fpm `.gitignore` rules were ignoring parts of TOML Fortran's source code (see [here](https://github.com/toml-f/toml-f/tree/main/src/tomlf/build)), therefore I usually settled for:

```gitignore
/build*/
/_*/
```

to ignore `build` and other `_build` directories.

Reply:
Re: removing the original .gitignore made by the "new" subcommand:

Well, I thought it was cleaner and more reliable and liked that it reduced any changes to the user directories, and was going to argue that but sampling a number of fpm projects made me change my mind so here is a proposed change that is admittedly  a compromise but was the best I could think of. Additional background is in the commit log.

Reply:
I have to admit that the most useful part of the `.gitignore` above is ignoring the build directories. Ignoring build artifacts by file extension is only meaningful with hand crafted Makefiles, autoconf or CMake which are used for in-tree builds. At least with CMake this is mainly happening by accident to me.

Now that I have tested it a bit I think the toplevel `.gitignore` might not be needed at all. The only argument in favor of a `.gitignore` is to make a user aware of the existence of this feature in `git` when using `fpm`.

Reply:
I am OK with that too, although that is an interesting list. So the new subcommand no longer creates a .gitignore.